### PR TITLE
User-story/1094-as-a-developer,-i-want-to-create-otanalytics

### PR DESCRIPTION
**Pull request by briemla**

### CODE REVIEW

File: OTAnalytics/application/datastore.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from domain.track import Track, TrackRepository
+
+
+class TrackParser(ABC):
+    @abstractmethod
+    def parse(self, file: Path) -> list[Track]:
+        pass
+
+
+class Video:
```
Comment (randy-seng) [inline OTAnalytics/application/datastore.py:14]: Can we define it as a dataclass?

File: OTAnalytics/application/datastore.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    path: Path
+
+    def get_image(self) -> None:
+        pass
+
+
+class VideoRepository:
```
Comment (randy-seng) [inline OTAnalytics/application/datastore.py:21]: See comment in `TrackRepository` in line 29

File: OTAnalytics/ui/gui.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import tkinter
+from parser.otvision_parser import OttrkParser
+
+import customtkinter
+from application.datastore import Datastore
+from customtkinter import CTk, CTkButton
+
+
+class OTAnalyticsApplication:
+    datastore: Datastore
```
Comment (randy-seng) [inline OTAnalytics/ui/gui.py:10]: Same here, do we want the variables to be static?

**Review by randy-seng — COMMENTED** [review]
Looks good! I have a few questions~

File: OTAnalytics/domain/track.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+
+@dataclass(frozen=True)
+class Track:
+    id: int
+    detections: list[Detection]
+
+
+class TrackRepository:
+    tracks: dict[int, Track] = {}
```
[the remark this answers is not in the document]
Comment (briemla) [reply OTAnalytics/domain/track.py:29]: You are right. I am a plain old java developer fighting with a snake...

**Review by randy-seng — APPROVED** [review]

**Pull request merged**